python/lsst/obs/hsc/transformRegistry.py

```diff
@@ -247,20 +255,75 @@ class HscDistortionConfig(Config):
                                    -1.03493809035e-37,
                                    8.09745928121e-38,
                                    ])
-    tolerance = Field(dtype=float, default=5.0e-3, doc="Tolerance for inversion (pixels)") # Much less than 1
-    maxIter = Field(dtype=int, default=10, doc="Maximum iterations for inversion") # Usually sufficient
+    tolerance = Field(dtype=float, default=5.0e-3, doc="Tolerance for inversion (pixels)")  # Much less than 1
+    maxIter = Field(dtype=int, default=10, doc="Maximum iterations for inversion")  # Usually sufficient
     plateScale = Field(dtype=float, default=1.0, doc="Plate scale (arcsec/mm)")
 
 
+def makeAstPolyMapCoeffs(order, xCoeffs, yCoeffs):
+    """Convert polynomial coefficients in HSC format to AST PolyMap format
+
+    Paramaters
+    ----------
+    order: int
+        Polynomial order
+    xCoeffs, yCoeffs: list of float
+        Forward or inverse polynomial coefficients for the x and y axes
+        of output, in this order:
+            x0y0, x0y1, ...x0yN, x1y0, x1y1, ...x1yN-1, ...
+        where N is the polynomial order.
+
+    Returns
+    -------
+    Forward or inverse coefficients for astshim.PolyMap
+    as a 2-d numpy array.
+    """
+    nCoeffs = (order + 1) * (order + 2) // 2
+    if len(xCoeffs) != nCoeffs:
+        raise ValueError("found %s xCcdToSky params; need %s" % (len(xCoeffs), nCoeffs))
+    if len(yCoeffs) != nCoeffs:
+        raise ValueError("found %s yCcdToSky params; need %s" % (len(yCoeffs), nCoeffs))
+
+    coeffs = np.zeros([nCoeffs * 2, 4])
+    i = 0
+    for nx in range(order + 1):
+        for ny in range(order + 1 - nx):
+            coeffs[i] = [xCoeffs[i], 1, nx, ny]
+            coeffs[i + nCoeffs] = [yCoeffs[i], 2, nx, ny]
+            i += 1
+    assert i == nCoeffs
+    return coeffs
+
+
 def makeHscDistortion(config):
-    """ Make an HscDistortion object
-    @param[in] config: pexConfig.Config object containing the distortion parameters needed to construct the
-    transform
+    """Make an HSC distortion transform
+
+    Note that inverse coefficients provided, but they are not accurate enough
+    to use: test_distortion.py reports an error of 2.8 pixels
+    (HSC uses pixels for its focal plane units) when transforming
+    from pupil to focal plane. That explains why the original HSC model uses
+    the inverse coefficients in conjunction with iteration.
+
+    Parameters
+    ----------
+    config: HscDistortionConfig
```

Comment:
config : `lsst.obs.subaru.HscDistortionConfig`

It's important to fully-qualify the type because it may be used in a different context.